locksmith/__tests__/token_middleware.test.js

```diff
@@ -0,0 +1,163 @@
+var httpMocks = require('node-mocks-http')
+var tokenMiddleware = require('../src/token_middleware')
+
+let request, response, validAuthorizationHeader
+
+beforeAll(() => {
+  request = httpMocks.createRequest()
+  response = httpMocks.createResponse()
+  validAuthorizationHeader = {
+    'Authorization':
+      'Bearer eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJsb2NrIjp7ImFkZHJlc3MiOiJqcWE2ZG5wMSIsIm5hbWUiOiJvaW9pb2kiLCJleHBpcmF0aW9uRHVyYXRpb24iOjI1OTIwMDAsImtleVByaWNlIjoiMTAwMDAwMDAwMDAwMDAwMDAiLCJtYXhOdW1iZXJPZktleXMiOjEwLCJvd25lciI6IjB4QWFBZEVFRDRjMEI4NjFjQjM2ZjRjRTAwNmE5QzkwQkEyRTQzZmRjMiJ9LCJpYXQiOjE1NDYxMzA4MzUsImV4cCI6MTU0NjEzMDgzOCwiaXNzIjoiMHhBYUFkRUVENGMwQjg2MWNCMzZmNGNFMDA2YTlDOTBCQTJFNDNmZGMyIn0.0xd36a7842eda1b2267c08b204407f9e71024a2152ae772882a1fa41a02696aee578688f02ef6c301451de95793bde669198359f0820fb35f6ba1303e89727a52601',
+  }
+})
+
+describe('Locksmith JWT Validation Middleware', () => {
+  describe('when a valid token is received', () => {
+    it('moves the request to the application', (done) => {
+      Date.now = jest.fn(() => 1546130835000)
+      var request = httpMocks.createRequest({ headers: validAuthorizationHeader, 
+        body: {'address':'jqa6dnp1','name':'oioioi','expirationDuration':2592000,
+          'keyPrice':'10000000000000000','maxNumberOfKeys':10,'owner':'0xAaAdEED4c0B861cB36f4cE006a9C90BA2E43fdc2'}})
+
+      tokenMiddleware(request, response, function next() {
+        expect(request.owner).toEqual('0xAaAdEED4c0B861cB36f4cE006a9C90BA2E43fdc2')
```

Comment:
nit - I prefer toBe over toEqual unless it is an array or object contents being compared since toEqual uses == and toBe uses ===. I always find it's toBe or not toBe. That is the question.